stem/model.py

```diff
@@ -289,10 +295,258 @@ def add_boundary_condition_by_geometry_ids(self, ndim_boundary: int, geometry_id
+        new_mesh.nodes = {
+            nn: process_model_part.mesh.nodes[nn] for nn in filtered_node_ids
+        }
+        new_mesh.elements = {}
+        return new_mesh
+
+    def add_random_field(
+            self, part_name: str, property_name: str, cov:float,
+            v_scale_fluctuation: float, anisotropy: List[float], angle: List[float], model_name: str = "Gaussian",
+            seed: int = 14, v_dim: int = 1, json_fname: Optional[str] = None
+    ):
+        """
+        Add a random field generator to perform random field for the given variable and the given model part.
+        The mean of the random field is taken from the material parameter.
+
+        Args:
+            - part_name (str): model of the part name where to apply the random field generation.
+            - property_name (str): property (or variable) name to generate the random field for (e.g. YOUNG_MODULUS).
+            - model_name (str): Name of the model to be used. Options are: "Gaussian", "Exponential", "Matern", "Linear"
+            - cov (float): The coefficient of variation of the random field.
+            - v_scale_fluctuation (float): The vertical scale of fluctuation of the random field.
+            - anisotropy (list): The anisotropy of the random field in the other directions (per dimension).
+            - angle (list): The angle of the random field (per dimension).
+            - seed (int): The seed number for the random number generator.
+            - v_dim (int): The dimension of the vertical scale of fluctuation. # TODO: make global variable
+
+        Raises:
+            - ValueError: if the model_name is not an invalid, implemented model.
+        """
+
+        # Check if the model part exists and retrieve the part
+        trgt_part = self.__get_model_part_by_name(part_name=part_name)
+
+        # Check if the model part is a body model part
+        if not isinstance(trgt_part, BodyModelPart):
+            raise ValueError(f"The target part, `{part_name}`, is not a body model part.")
+
+        # Check if the material of the body model part is soil
+        if trgt_part.material is None:
+            raise ValueError(f"No material assigned to the body model part!")
+
+        if not trgt_part.material.is_property_in_material(property_name=property_name):
+            raise ValueError(f"The property for which a random field needs to be generated, `{property_name}`, "
+                             f"is not part of the soil material.")
+
+        # Get the property of the soil material, this is the mean value of the random field.
+        # Checks also if the material of the body model part is soil contains the desired parameter
+        mean_value = trgt_part.material.get_property_in_material(property_name=property_name)
+        if isinstance(mean_value, bool) or not isinstance(mean_value, (float, int)):
+            raise ValueError(f"The property for which a random field needs to be generated, `{property_name}`, "
```

Comment:
The check is required because apparently a boolean is an integer but an integer is not a boolean:
isinstance(1, bool)
Out[7]: False
isinstance(True, int)
Out[8]: True